app/src/main/java/com/projectthrive/chess/ui/main/MainFragment.kt

```diff
@@ -24,11 +25,20 @@ class MainFragment : Fragment() {
         val rootView = inflater.inflate(R.layout.main_fragment, container, false)
         val mainBoard = rootView.findViewById<GridLayout>(R.id.main_board)
 
+        viewModel = GameModel()
+        val pieces = viewModel.initialPiecesSetup()
+        piecesInGame.postValue(pieces)
+
         for (i in 0..7) {
             for (j in 0..7) {
                 val squareView =
                     layoutInflater.inflate(R.layout.square_view, mainBoard, false) as ImageView
-                squareView.setImageDrawable(getColoredSquare(i, j))
+                if (pieces.containsKey(Position(i,j))) {
```

Comment:
This should not be done on the onCreateView() method, but instead at any time when piecesInGame is updated.